worker/util/network.py

```diff
@@ -0,0 +1,41 @@
+"""A module for networking related utilities."""
+
+import random
+import socket
+
+
+def get_local_ip():
+    """
+    Get the local IP address.
+
+    Equivalent to https://github.com/stencila/executa/blob/753207cb31298578497d978265c718e20b583a05/src/tcp/util.ts#L15
+    Thanks to https://stackoverflow.com/questions/166506/finding-local-ip-addresses-using-pythons-stdlib
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    try:
+        s.connect(("8.8.8.8", 80))
```

Comment:
This is so haxx and I hate that we have to do things like this. I'd much prefer having the user specify an IP address  if possible.